libsmtutil/Z3CHCInterface.cpp

```diff
@@ -95,9 +95,12 @@ pair<CheckResult, CHCSolverInterface::CexGraph> Z3CHCInterface::query(Expression
 		}
 		// TODO retrieve model / invariants
 	}
-	catch (z3::exception const&)
+	catch (z3::exception const& _err)
 	{
-		result = CheckResult::ERROR;
+		if (_err.msg() == string("max. resource limit exceeded"))
+			result = CheckResult::UNKNOWN;
```

Comment:
Can we display a warning for "not solvable yet"?

Reply:
That's basically what unknown means.
The warning should be emitted by CHC though, not the solver interface.

Reply:
Also: we do display a warning like that, it's the "... might happen"